contracts/arbitrage/RevoUniswapArbitrage.sol

```diff
@@ -0,0 +1,385 @@
+        );
+        uint256 _halfStartTokens = _args.amountZapToken / 2;
+
+        for (uint256 i = 0; i < _args.paths.length; i++) {
+            IERC20 _liquidityToken = IERC20(
+                _args.paths[i][_args.paths[i].length - 1]
+            );
+
+            _liquidityTokenBalances[i] = UniswapRouter.swap(
+                _farmBot.swapRouter(),
+                _args.paths[i],
+                _halfStartTokens,
+                IERC20(_args.zapToken),
+                _args.amountsOutMin[i],
+                _args.deadline
+            );
+
+            // Pre-approve the broker to spend the liquidity constituent token we swapped for
+            _liquidityToken.safeIncreaseAllowance(
+                _args.brokerAddress,
+                _liquidityTokenBalances[i]
+            );
+        }
+
+        LiquidityAmounts memory _liquidityAmounts = LiquidityAmounts(
+            _liquidityTokenBalances[0],
+            _liquidityTokenBalances[1],
+            _args.minAmountStakingToken0,
+            _args.minAmountStakingToken1
+        );
+
+        uint256 _initialFpBalance = IERC20(_args.farmBotAddress).balanceOf(
+            address(this)
+        );
+        // Mint FP using the broker; the broker will send this contract some FP and any remaining stakingToken0 and stakingToken1.
+        // any stakingToken0 and stakingToken1 returned by the broker should be sent back to the sender
+        RevoFPBroker(_args.brokerAddress).getUniswapLPAndDeposit(
+            _args.farmBotAddress,
+            _liquidityAmounts,
+            _args.deadline
+        );
+
+        // Swap the RFP for the original starting token. The caller specifies the minimum deisred amount in _args.minZapTokenOut.
+        // Since the swap will fail if the output amount is below this value, the caller is guaranteed a profit if _args.minZapTokenOut
+        // is greater than _args.amountZapToken and the function returns.
+        uint256 _amountZapTokenOut = UniswapRouter.swap(
+            _farmBot.swapRouter(),
+            _args.zapPath,
+            IERC20(_args.farmBotAddress).balanceOf(address(this)) -
+                _initialFpBalance,
+            IERC20(_args.farmBotAddress),
+            _args.minZapTokenOut,
+            _args.deadline
+        );
+
+        // Extra safety check; should not be strictly necessary if the swap router is honest
+        require(
+            _amountZapTokenOut >= _args.minZapTokenOut,
+            "RevoUniswapArbitrage: Arbitrage is not profitable"
+        );
+
+        // Send original token back to sender, along with any constituent LP tokens that went unused by the broker
+        IERC20(_args.zapToken).safeTransfer(msg.sender, _amountZapTokenOut);
+        for (uint256 i = 0; i < _args.paths.length; i++) {
+            IERC20 _liquidityToken = IERC20(
+                _args.paths[i][_args.paths[i].length - 1]
+            );
+            uint256 _amountReturnedLiquidityToken = IERC20(_liquidityToken)
```

Comment:
nit: do you need IERC20(_liquidityToken) ? could you just use _liquidityToken since you already wrap it in IERC20(...) in the prev line?